connect/api/src/main/java/org/apache/kafka/connect/connector/package-info.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Provides pluggable interfaces for Connector and Task implementations.
```

Comment:
Nit: is "pluggable" really necessary here? I've always used that term to identify parts of a code base or application that can be swapped out for something else, but only on an opt-in basis, and with a reasonable default available out of the box.

Since there is no default `Connector` or `Task` implementation and it's required to write one in order to run a Kafka Connect connector, I'd opt to leave this bit out.

Also worth noting that the `ConnectorContext` and `ConnectRecord` classes aren't really pluggable; they're defined by the Connect framework and users have no chance to swap in their own implementation (at least, not in a way that alters how they are used by the Connect framework).